libs/utils/analysis_tool.py

```diff
@@ -0,0 +1,116 @@
+                           to parse specific options if any (optional).
+    :type options_parser: ArgumentParser
+    """
+
+    def __init__(self, options_parser=None):
+        self.output = None
+        self.options_parser = options_parser
+
+    def _setup_logging(self):
+        """
+        Setup logging for the analysis tool.
+
+        Analysis tools shouldn't output to standard output as their
+        output can be used by other tools. Configure the logger to
+        output only error messages
+        """
+        self._log = logging.getLogger('AnalysisTool')
+        level = logging.INFO if self.args.log_stdout else logging.ERROR
+        LisaLogging.setup(level=level)
+
+    def _parse_args(self):
+        """
+        Parse command line arguments given to the tool, also use the parser
+        object if one is passed by the more specific class so that the
+        tool specific options are also parsed
+        """
+        parser = self.options_parser
+
+        if not parser:
+            parser = argparse.ArgumentParser(
+                    description='LISA Analysis tool configuration')
+
+        # General options that all tools accept
+        parser.add_argument('--trace', type=str,
+                default=None,
+                help='Path of the trace file or directory to be used',
+                required=True)
+
+        parser.add_argument('--log-stdout', action='store_true',
+                default=False,
+                help='Output logging to stdout',
+                required=False)
+
+        self.args = parser.parse_args()
+
+    def run_analysis(self):
+        """
+        Run an analysis and store results
+
+        The specific analysis tool should override this and perform
+        analysis and provide results in self.output
+        """
+        raise NotImplementedError('error')
+
+    def run_output(self):
+        """
+        Present output results to the user, an AnalysisTool should override
+        this to provide custom output formatting. The default is just to
+        print the results to stdout.
+        """
+        if not self.output:
+            raise RuntimeError('Analysis tool is expected to generate results')
+        print self.output
+
+    def run_main(self):
+        """
+        The main function that sets up, runs the analysis and provides
+        output to the user.
+
+        The specific analysis tool should call this function to start the analysis
+        which will perform analysis and report the result. The tool should also
+        override run_analysis to provide custom analysis and optionally override
+        run_output to provide custom output.
+        """
+
+        self._parse_args()
+        self._setup_logging()
+
+        # Create trace object
+        self._log.info('Creating trace object...')
+        self.trace = Trace(None, self.args.trace)
+
+        self.run_analysis()
+        self.run_output()
```

Comment:
Now every script produce its output in a single `self.output` variable, which is printed by the default (dummy) implementation of `run_output()`... if in the future you should need any "special formatting" it's quite likely you still need to update all scripts because the simple `self.output` is... too simple.

Do you have any special use-case already in mind? In that case you should provide an example, otherwise your arguments now are not convincing to me.

Regarding `AnalysisTool::run_analysis` vs `AnalysisTool::analyse` I cannot see why you are so much into to that name?
Is there any other reason behind just the name itself?
Otherwise I think we should use @bjackman's vote on this as well ;-)